cpp/src_prims/timeSeries/stationarity.h

```diff
@@ -0,0 +1,414 @@
+  // Table 1, Kwiatkowski 1992
+  const DataT crit_vals[4] = {0.347, 0.463, 0.574, 0.739};
+  const DataT pvals[4] = {0.10, 0.05, 0.025, 0.01};
+
+  IdxT i = blockIdx.x * blockDim.x + threadIdx.x;
+
+  if (i < n_batches) {
+    DataT s2Ai = s2A[i];
+    DataT etai = eta[i];
+    DataT s2Bi = s2B[i];
+
+    s2Ai /= n_samples_f;
+    etai /= n_samples_f * n_samples_f;
+
+    DataT kpss_stat = etai / (s2Ai + s2Bi);
+
+    // Interpolate the pvalue (y) based on the kpss stat (x)
+    DataT pvalue = pvals[0];
+#pragma unroll
+    for (IdxT k = 0; k < 3; k++) {
+      if (kpss_stat >= crit_vals[k] && kpss_stat < crit_vals[k + 1]) {
+        pvalue = pvals[k] + (pvals[k + 1] - pvals[k]) *
+                              (kpss_stat - crit_vals[k]) /
+                              (crit_vals[k + 1] - crit_vals[k]);
+      }
+    }
+    if (kpss_stat >= crit_vals[3]) {
+      pvalue = pvals[3];
+    }
+
+    // A higher pvalue means a higher chance that the data is stationary
+    results[i] = (pvalue > pval_threshold);
+  }
+}
+
+/* A structure that defines a function to get the column of an element of
+ * a matrix from its index. This makes possible a 2d scan with thrust.
+ * Found in thrust/examples/scan_matrix_by_rows.cu
+ */
+template <typename IdxT>
+struct which_col : thrust::unary_function<IdxT, IdxT> {
+  IdxT col_length;
+  __host__ __device__ which_col(IdxT col_length_) : col_length(col_length_) {}
+  __host__ __device__ IdxT operator()(IdxT idx) const {
+    return idx / col_length;
+  }
+};
+
+/**
+ * @brief Applies the stationarity test to the given series
+ * 
+ * @details The following algorithm is based on Kwiatkowski 1992:
+ *          - Center each series around its mean
+ *          - Calculate s^2 (eq. 10) and eta (eq. 11)
+ *          - Deduce the p-value and compare against the threshold
+ * 
+ * @note The data is a column-major matrix where the series are columns.
+ *       This function will be called at most twice by `stationarity`
+ *
+ * @tparam      DataT           Scalar type of the data (float or double)
+ * @tparam      IdxT            Integer type of the indices
+ * @param[in]   y_d             Input data
+ * @param[out]  results         Boolean array to store the results of the test
+ * @param[in]   n_batches       Number of batches
+ * @param[in]   n_samples       Number of samples
+ * @param[in]   allocator       cuML device memory allocator
+ * @param[in]   stream          CUDA stream
+ * @param[in]   cublas_handle   cuBLAS handle
+ * @param[in]   pval_threshold  P-value threshold above which a series is
+ *                              considered stationary 
+ */
+template <typename DataT, typename IdxT>
+static void _is_stationary(const DataT* y_d, bool* results, IdxT n_batches,
+                           IdxT n_samples,
+                           std::shared_ptr<MLCommon::deviceAllocator> allocator,
+                           cudaStream_t stream, cublasHandle_t cublas_handle,
```

Comment:
this function doesn't seem to need `cublas_handle` arg?

Reply:
Indeed. I've removed the cublasHandle from stationarity entirely.